ca/ca_test.go

```diff
@@ -115,9 +115,11 @@ var (
 	OIDExtensionCTPoison = asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 11129, 2, 4, 3}
 
 	issuanceModes = []IssuanceMode{
-		{name: "non-precertificate", usePrecertificateFlow: false, enablePrecertificateFlow: false},
-		{name: "precertificate", usePrecertificateFlow: true, enablePrecertificateFlow: true},
-		{name: "precertificate-disabled", usePrecertificateFlow: true, enablePrecertificateFlow: false},
+		{name: "non-precertificate", issuePrecertificate: false, issueCertificateForPrecertificate: false, enablePrecertificateFlow: false},
+		{name: "precertificate", issuePrecertificate: true, issueCertificateForPrecertificate: false, enablePrecertificateFlow: true},
+		{name: "precertificate-disabled", issuePrecertificate: true, issueCertificateForPrecertificate: false, enablePrecertificateFlow: false},
+		{name: "certificate-for-precertificate", issuePrecertificate: false, issueCertificateForPrecertificate: true, enablePrecertificateFlow: true},
+		{name: "certificate-for-precertificate-disabled", issuePrecertificate: false, issueCertificateForPrecertificate: true, enablePrecertificateFlow: false},
```

Comment:
@cpu Thanks! I think there's not much value in adding the test. I spent some time investigating what would be required to add that, and it turned out that it made things much more messy than we might expect. One reason is that each CA object maintains counters that are checked in the tests, and adding the new mode would mean making the counter-checking logic (even) more complicated.

I added a new commit that adds some comments about the structure of this test, including issues related to this decision.